crates/oxc_linter/src/rules/eslint/preserve_caught_error.rs

```diff
@@ -0,0 +1,591 @@
+use oxc_ast::AstKind;
+use oxc_ast::ast::{
+    Argument, BindingPattern, CatchClause, Expression, ObjectExpression, ObjectPropertyKind,
+    PropertyKey, Statement, ThrowStatement, TryStatement,
+};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_semantic::IsGlobalReference;
+use oxc_span::Span;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use crate::{AstNode, context::LintContext, rule::Rule};
+
+fn preserve_caught_error_diagnostic(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("preserve-caught-error")
+        .with_help(
+            "When re-throwing an error, preserve the original error using the 'cause' property",
+        )
+        .with_label(span)
+}
+
+fn missing_catch_parameter_diagnostic(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("preserve-caught-error")
+        .with_help("Catch clause must have a parameter when 'requireCatchParameter' is enabled")
+        .with_label(span)
+}
+
+#[derive(Debug, Default, Clone, Serialize, Deserialize, JsonSchema)]
+#[schemars(rename_all = "camelCase")]
+#[serde(rename_all = "camelCase")]
+struct ConfigElement0 {
+    require_catch_parameter: bool,
+}
+
+#[derive(Debug, Default, Clone, Serialize, Deserialize, JsonSchema)]
+pub struct PreserveCaughtError(ConfigElement0);
+
+// Helper functions - global scope for better reusability and testing
+
+// Recursively check statements for throw expressions
+fn check_statement<'a>(
```

Comment:
I think we probably want to use a visitor here rather than this manual visit implementation.

e.g. I don't think

```
try {} catch (e) {
    {
        throw new Error('oops')
    }
}
```

would be reported since the throw stmt is within a block stmt which you don't visit. Look at
https://github.com/oxc-project/oxc/blob/b794af1924c62f134f17a2866f374709d514195c/crates/oxc_linter/src/rules/eslint/no_extra_bind.rs#L106-L116
for an example